src/plugins/shared/clusterOptions/options/selectExistingClusterOption.ts

```diff
@@ -0,0 +1,102 @@
+import * as vscode from "vscode";
+import { ReadyAzureSessionProvider } from "../../../../auth/types";
+import { ClusterPreference } from "../../types";
+import { selectSubscriptions } from "../../../../commands/aksAccount/aksAccount";
+import { getClusters, getManagedCluster, getKubeconfigYaml, Cluster } from "../../../../commands/utils/clusters";
+import { getFilteredSubscriptions } from "../../../../commands/utils/config";
+import { Errorable, failed } from "../../../../commands/utils/errorable";
+import { longRunning } from "../../../../commands/utils/host";
+import { RecentCluster } from "../state/recentCluster";
+
+
+export async function selectExistingClusterOption(sessionProvider: ReadyAzureSessionProvider): Promise<Errorable<ClusterPreference>> {
+    // allow user to select subscriptions
+    await selectSubscriptions();
+
+    // get subscriptions
+    const subscriptions = getFilteredSubscriptions();
```

Comment:
💡 Error handling:

Key aspect here might be we need to add if missing is error_handlers. I can checkout the code and might see what's going on through this.

Reply:
Yes good point, updated to handle if users don't have any subscriptions